src/components/new-safe/steps/Step4/useSafeCreation.ts

```diff
@@ -0,0 +1,100 @@
+import type { Dispatch, SetStateAction } from 'react'
+import { useCallback, useEffect, useState } from 'react'
+import { useWeb3 } from '@/hooks/wallets/web3'
+import { useCurrentChain } from '@/hooks/useChains'
+import useWallet from '@/hooks/wallets/useWallet'
+import type { EthersError } from '@/utils/ethers-utils'
+import type { PendingSafeData } from '@/components/new-safe/steps/Step4/index'
+import type { PendingSafeTx } from '@/components/create-safe/types'
+import {
+  checkSafeCreationTx,
+  createNewSafe,
+  getSafeCreationTxInfo,
+  getSafeDeployProps,
+  handleSafeCreationError,
+} from '@/components/new-safe/steps/Step4/logic'
+
+export enum SafeCreationStatus {
+  AWAITING = 1,
+  PROCESSING = 2,
+  WALLET_REJECTED = 3,
+  ERROR = 4,
+  REVERTED = 5,
+  TIMEOUT = 6,
+  SUCCESS = 7,
+  INDEXED = 8,
+  INDEX_FAILED = 9,
+}
```

Comment:
Can you add a comment explaining that these reference steps? On intial sight I was confused as to why we assign a value directly and not use the default from `0`.

Reply:
Good point, we don't need to explicitely set the numbers as its fine to start at 0